### PR TITLE
Activate: Store creation incomplete notification

### LINKED ISSUE #9088: Send a scheduled notification after 24hrs to those users that entered store name but did not subscribe to the free trial.

### PR DESCRIPTION
**Pull request by 0nko**
Fixes #9088, a subtask of #8999.

This PR schedules a notification 24 hours after a store name is entered during store creation and then it's abandoned without completion. When the notification is tapped, the store creation is restarted with the original name pre-filled.

[device-2023-05-24-010103.webm](https://github.com/woocommerce/woocommerce-android/assets/1522856/5799439e-5a9c-4290-a335-1232093726e4)

**To test:**

_Test 1_

1. From a logged-out state, tap on Get Started CTA and begin the store creation process
2. When you enter the store name and tap Continue, minimize the app
3. Wait for 24 hours
4. **Verify that a notification is displayed**
5. Tap on it
6. **Verify that the store name picker is shown with the original name pre-filled**
7. Tap on the back button and **make sure the back stack is looking normal** (no duplicates or anything weird)

_Test 2_

1. Log in
2. Go to More menu -> Settings -> Site picker
3. Repeat steps 1-7 from Test 1
4. Verify that everything is working

> **Note**
> If you want to make the testing go quicker, you can change the notification units to seconds [here](https://github.com/woocommerce/woocommerce-android/compare/issue/9060-store-ready-notification...issue/9088-store-creation-incomplete?expand=1#diff-74c116a16145d5ea7d01758d98557b5a20c465d60e70c9e45b59373b810b057bR45). 🙂

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/summary/StoreCreationSummaryViewModel.kt [hunk 1]
```diff
@@ -92,6 +93,9 @@ class StoreCreationSummaryViewModel @Inject constructor(
                 accountStore.account.userName
             localNotificationScheduler.scheduleNotification(StoreCreationFinishedNotification(name))
         }
+
+        // No need to display a notification to complete store creation anymore
+        localNotificationScheduler.cancelScheduledNotification(STORE_CREATION_INCOMPLETE)
```
Comment (hafizrahman) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/summary/StoreCreationSummaryViewModel.kt:98]: Awesome that you realized this use case!

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt [hunk 2]
```diff
@@ -751,6 +755,22 @@ class MainActivity :
         observeBottomBarState()
     }
 
+    private fun shortcutOpenStoreCreation(storeName: String?) {
```
Comment (hafizrahman) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt:758]: nitpick: maybe rename to `openShortcutToStoreCreation()` to follow "use verb for function name".

Comment (hafizrahman) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt:758]: I just noticed that there's already existing functions that use the same `shortcutXXX()` pattern, so feel free to ignore the above to keep with the pattern instead.

**Review by hafizrahman — APPROVED** [review]
This works well in my test. Awesome work! I checked the commit one by one to get a better understanding how setting up a scheduled local notification will work, and things look good to me.

I left a non-blocking comment so feel free to merge after reading that and fixing conflicts.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt (hunk 2 above)
Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt:758]: Yeah, I just copied what was already there, otherwise I'd agree 💯

**Comment by 0nko** [comment]
Thanks @hafizrahman!

**Pull request merged**